refactor: Add header types

danoro96 opened the pull request:
Closes #13587, #13588, #13589, #13585

Add types for Header, HeaderGlobalBar, HeaderMenu, HeaderGlobalAction
#### Changelog

**Changed**

- Added Header.tsx
- Added HeaderGlobalBar.tsx
- Added HeaderMenu.tsx
- Added HeaderGlobalAction.tsx

#### Testing / Reviewing

Fire up StoryBook and check that everything works as expected

dependabot[bot] committed:
chore(deps): bump browserify-sign (#15042)

Bumps [browserify-sign](https://github.com/crypto-browserify/browserify-sign) from 4.2.1 to 4.2.2.
- [Changelog](https://github.com/browserify/browserify-sign/blob/main/CHANGELOG.md)
- [Commits](https://github.com/crypto-browserify/browserify-sign/compare/v4.2.1...v4.2.2)

---
updated-dependencies:
- dependency-name: browserify-sign
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>
Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>

lewandom committed:
fix(icons-react): use declare keyword in bucket TypeScript types (#15094)

dependabot[bot] committed:
chore(deps): bump lodash in /packages/icons-react/examples/storybook (#15010)

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.11 to 4.17.21.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.11...4.17.21)

---
updated-dependencies:
- dependency-name: lodash
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>
Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>

renovate[bot] committed:
chore(deps): update commitlint monorepo to v18 (major) (#15105)

* chore(deps): update commitlint monorepo to v18

* chore: yarn dedupe

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>

carbon-automation[bot] committed:
chore(release): v11.42.0-rc.0 (#15112)

Co-authored-by: andreancardona <andreancardona@users.noreply.github.com>

tw15egan committed:
docs(FluidMultiSelect): update propType definitions for FluidMultiSelect (#15108)

* docs(FluidMultiSelect): update propType definitions for FluidMultiSelect

* test(snapshot): update snapshots

tw15egan committed:
style(Slug): increase touch target to 24px for mini, 2xs variants (#15100)

Nirajsah committed:
feat(component): replaced  ->  component (#15084)

tw15egan committed:
feat(gradient): linear gradient mixin (#15081)

* feat(gradient): add linear gradient mixin

* feat(gradient): add gradient tokens, update demo story

* fix(gradient): change default direction

* chore(snapshot): update snapshots

* refactor(Gradient): use color tokens, update token names

* test(snapshot): update snapshots

* docs(ai-gradient): mark mixin experimental, update year

* chore(storybook): remove test story

tay1orjones committed:
test(e2e): add proper tags to avt tests (#14915)

      * test(e2e): add proper tags to avt tests

* ci(avt): shard avt tests

* fix: typo

* test(config): increase playwright expect timeout

* test(playwright): increase timeout config

* test(progressindicator): update to use proper tags

* test(progressindicator): correct focus order

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>
Co-authored-by: TJ Egan <tw15egan@gmail.com>

lewandom committed:
fix(icons-react): avoid special tokens in bucket .d.ts exports (#15118)

guidari committed:
Added trap focus and focus redirection (#14877)

* fix: added trap focus and focus redirection

* test: added test to match tabIndex

* fix: fixed focus issues to action button

* chore: remove test story

* fix: fixed test

* fix: fixed avt test to match component

---------

Co-authored-by: TJ Egan <tw15egan@gmail.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>
Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>

dudley-ibm committed:
feat(Pictograms): new artwork for three watsonx pictograms (#15116)

tay1orjones committed:
chore(monorepo): update to yarn 4 (#15127)

* ci(cache): turn off caching in the merge queue

* chore(project): update to yarn 4

* chore: output of yarn install

carbon-automation[bot] committed:
chore(release): v11.42.0 (#15134)

Co-authored-by: tay1orjones <tay1orjones@users.noreply.github.com>

tw15egan committed:
refactor(NumberInput): remove pseudo elements and use box-shadow (#15128)

Tresau committed:
fix: BreadcrumbItem aria-current PropType fix (#15052)

* feat: add types for Breadcrumbs

* revert: undo PropType changes and supress TS error instead

* fix: update 'aria-current' to match standards

The PropType definition for 'aria-current' is to broad.
Update the PropType to match AriaAttributes['aria-current']

BREAKING CHANGE: PropType is changed to be more specific

* fix: update snapshot

alisonjoseph committed:
docs: remove cra-template (#15124)

alisonjoseph committed:
docs: remove codesandbox and codesandbox-with-sass-compilation examples (#15122)

* docs: remove codesandbox and codesandbox-with-sass-compilation examples

* docs: remove codesandbox and codesandbox-with-sass-compilation examples

* chore: yarn

* chore: fix merge

---------

Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>

conradennis committed:
feat(icons): new icons september thru november (#15125)

* feat(icons): new icons september thru november

Submitting new and updated UI icons from September through November 2023
all updates and additions approved by BXD

Note: please hide the two deprecated icons from the Carbon/IDL library

[70] net new icons
[1] updated icons
[8] icons moved to new category
-
[1] new icon category
-
[2] deprecated icons

updated categories.yml with additions and updates
updated deprecated.yml with deprecated file names
updated ui-icon-master.ai with additions and updates

@lmapes to update friendly names and aliases in an additional PR

* chore(test): update snapshots

* chore: update snapshots

* updated meta data for Carbon for Coud

changed friendly name for UI icon ibm-cloud--pal from IBM Cloud Pattern and Asset Library to Carbon for Cloud

updated aliases in icons.yml

* Update to icon ibm-cloud--pal artwork

Mike noticed the SVG artwork was wonky. Submitting the updated artwork.

---------

Co-authored-by: TJ Egan <tw15egan@gmail.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>

guidari committed:
test: added avt keyboard test (#15149)

guidari committed:
test: added avt keyboard nav (#15143)

dependabot[bot] committed:
chore(deps): bump chromedriver from 91.0.1 to 119.0.1 (#15140)

* chore(deps): bump chromedriver from 91.0.1 to 119.0.1

Bumps [chromedriver](https://github.com/giggio/node-chromedriver) from 91.0.1 to 119.0.1.
- [Commits](https://github.com/giggio/node-chromedriver/compare/91.0.1...119.0.1)

---
updated-dependencies:
- dependency-name: chromedriver
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(yarn): run dedupe

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: TJ Egan <tw15egan@gmail.com>

Nirajsah committed:
feat(component): replaced  ->  component (#15120)

renovate[bot] committed:
fix(deps): update dependency downshift to v8.2.3 (#15156)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

guidari committed:
Add Keyboard Nav AVT test for `FluidTextArea` (#15161)

* test: changed text

* test: added avt keyboard test

* test: fixed test

guidari committed:
test: added avt keyboard test (#15151)

dudley-ibm committed:
feat(Pictograms): new pictograms and categories v21 (#15135)

* feat(Pictograms): new pictograms and categories v21

* chore(snapshots): update snapshots

* fix(Pictograms): rename cloud--pal to carbon--for-cloud

* docs(Pictograms): update alias

---------

Co-authored-by: TJ Egan <tw15egan@gmail.com>

tay1orjones committed:
fix(sidenavitems): remove conditional so children are always cloned (#15205)

* fix(sidenavitems): remove conditional so children are always cloned

* chore: remove test code

guidari committed:
test: added avt keyboard test (#15164)

guidari committed:
Add Keyboard Nav AVT test for `FluidSelect` (#15153)

* test: added avt keyboard test

* chore: fixed date for copyright

guidari committed:
feat: added $layer token in the background-color (#14726)

* feat: added layer toke in the background color

* fix: remove scss

* fix: fixed vrt test name

* fix: added background style

* fix: removed storybook and vrt testing

* fix: added layer stories

* fix: removed wrong files

---------

Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>
Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

tw15egan committed:
feat(Slug): Add `slug` prop to `TextInput`, `TextArea`, `NumberInput`, and `DatePicker`, add revert functionality to `Slug` (#15115)

* feat(TextInput): support slug prop

* fix(TextInput): add types for slug prop

* refactor(Slug): remove old file, move demo to form story

* docs(Storybook): update form story

* feat(NumberInput): add slug to NumberInput

* refactor(NumberInput): remove pseudo elements and use box-shadow

* style(NumberInput): fix issue with z-index, add invalid styles

* feat(DatePicker): add slug prop to DatePickerInput

* feat(TextArea): add slug prop to TextArea

* test(snapshot): update snapshots

* fix(TextArea): remove console warning

* style(Fluid): adjust styles for fluid inputs

* docs(Storybook): fix ids

* fix(Slug): adjust positioning, add storybook controls

* fix(Storybook): add story styles to form example

* fix(Slug): adjust positioning when invalid, warning state

* style(Slug): adjust padding when slug is present

* docs(Storybook): add examples with revert functionality

* fix(NumberInput): fix type error

* refactor(storybook): adjust stories

* feat(Slug): add revertLabel prop

* fix(Undo): replace undo, redo with coreect asset

* fix(Slug): adjust revert positioning, remove margin

* fix(NumberInput): don't set defaultValue if not provided, remove console

* chore(files): add ai-gradient to react package

* test(snapshot): update snapshots

guidari committed:
test: added avt keyboard test (#15166)

guidari committed:
Add Keyboard Nav AVT test for `FluidDatePicker` (#15141)

* test: added avt keyboard test

* fix: fixed skeleton test

* fix: fixed skeleton test

* fix: removed wrong word in the props

guidari committed:
Add Keyboard Nav AVT test for `FluidMultiSelect` (#15146)

* test: added avt keyboard test

* fix: fixed naming

* test: added test to condensed variant

* test: changed text

guidari committed:
Add Keyboard Nav AVT test for `FluidComboBox` (#15137)

* test: added avt keyboard test

* fix: fixed spelling

tw15egan committed:
fix(Slug): add slug to index.ts (#15216)

tw15egan committed:
feat(ListBox): add `slug` prop to `Dropdown`, `MultiSelect`, `ComboBox`, and `Select` (#15206)

* feat(TextInput): support slug prop

* fix(TextInput): add types for slug prop

* refactor(Slug): remove old file, move demo to form story

* docs(Storybook): update form story

* feat(NumberInput): add slug to NumberInput

* refactor(NumberInput): remove pseudo elements and use box-shadow

* style(NumberInput): fix issue with z-index, add invalid styles

* feat(DatePicker): add slug prop to DatePickerInput

* feat(TextArea): add slug prop to TextArea

* test(snapshot): update snapshots

* fix(TextArea): remove console warning

* style(Fluid): adjust styles for fluid inputs

* docs(Storybook): fix ids

* fix(Slug): adjust positioning, add storybook controls

* fix(Storybook): add story styles to form example

* fix(Slug): adjust positioning when invalid, warning state

* style(Slug): adjust padding when slug is present

* docs(Storybook): add examples with revert functionality

* fix(NumberInput): fix type error

* refactor(storybook): adjust stories

* feat(Slug): add revertLabel prop

* fix(Undo): replace undo, redo with coreect asset

* feat(ListBox): add Slug prop to ListBox components

* fix(Slug): adjust revert positioning, remove margin

* docs(Storybook): rearrange storybook order

* fix(NumberInput): don't set defaultValue if not provided, remove console

* feat(Select): add Slug prop to Select

* chore(files): add ai-gradient to react package

* test(snapshot): update snapshots

* refactor(Slug): remove aria-hidden

tay1orjones committed:
chore: update snapshots (#15224)

andreancardona committed:
feat: 14595-update-ally-test-tags (#14913)

Co-authored-by: TJ Egan <tw15egan@gmail.com>
Co-authored-by: Guilherme Datilio Ribeiro <guilhermedatilio@gmail.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>

Tomahaawk committed:
feat(Modal|ComposedModal): add inline loading option (#14490)

* feat(Modal|ComposedModal): add inline loading option

* refactor(Modal|ComposedModal): add default values

* refactor(Modal|ComposedModal): change loading button styles

* feat: update format

* fix(Modal): add missing ModalProps

* fix(Modal|ComposedModal): type errors

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

carbon-automation[bot] committed:
chore(release): v11.43.0-rc.0 (#15225)

Co-authored-by: tay1orjones <tay1orjones@users.noreply.github.com>

dudley-ibm committed:
feat(Pictograms): watson logo pictogram removed (#15221)

* feat(Pictograms): watson logo pictogram removed

* fix(pictograms): point to possible replacement picto

---------

Co-authored-by: Guilherme Datilio Ribeiro <guilhermedatilio@gmail.com>
Co-authored-by: TJ Egan <tw15egan@gmail.com>

sstrubberg committed:
Support docs revised (#15180)

* chore(strategic-adopters): added more of them

* docs(support): revised support guide

* docs(support): gh project docs

* docs(support): intro bits

* docs(support): bits and bobs

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* docs(support): worked on overview

* Update docs/guides/support.md

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>

* docs(support): brackets

* docs(support): fixed link

---------

Co-authored-by: Lauren Rice <43969356+laurenmrice@users.noreply.github.com>
Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

harukikuri committed:
fix(react): warnings from SideNav in console from inert prop (#14990) (#15047)

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: TJ Egan <tw15egan@gmail.com>

danoro96 committed:
refactor: add header types

k-rajat19 committed:
feat: added support for `Alt+Down Arrow` and `Alt+Up Arrow` in ComboBox (#14892)

* feat: added support

* feat: added test

* fix: tests

renovate[bot] committed:
chore(deps): update dependency lint-staged to v15 (#15158)

* chore(deps): update dependency lint-staged to v15

* chore: yarn dedupe

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>

janhassel committed:
feat(menu): support menu modes (#14793)

* feat(menu): support menu mdoes

* refactor(menu): use warning instead of error for valid use

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

shreeup committed:
Add TypeScript types to Switcher, SwitcherDivider, SwitcherItem #13584 (#14989)

* fix(react): convert js to typescript #13584

* fix(react): js -> ts -SwitcherItem ,Divider #13584

* fix(code): tests, proptype description #13584

* fix(code): warning #13584

* fix(code): deleted from storybook as per review

* fix(react): incorporate feedback

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

tay1orjones reviewed (changes requested):
Thanks for taking this on! I think it looks great. The only thing I see is that each interface needs the prop comments copied from the proptypes for future documentation efforts.

harukikuri committed:
fix(react): downshift replacing id prop in dropdown (#15241) (#15249)

tw15egan committed:
test(Slug): add VRT, AVT tests (#15238)

* test(Slug): add VRT, AVT tests

* test(Slug): test form story

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

renovate[bot] committed:
chore(deps): update yarn to v4.0.2 (#15228)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

guidari committed:
Added "as" prop to `Link` (#15233)

* feat: added as prop

* feat: added example

* fix: removed undefined from typescript

* fix: rollback change for className

* fix: added undefined

* feat: added as description

* fix: added control as false and updated snapshot

tw15egan committed:
chore(docs): remove `bx` references (#15217)

* docs(bx): remvoe old prefix from docs

* docs(bx): remove old prefix from docs

* fix(Storybook): remove fileuploader story files, revert readme changes

* fix(readme): fix all-contributors format

* fix(storybook): remove references to deleted file

* fix(FileUploader): remove deleted file reference

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

carbon-automation[bot] committed:
chore(release): v11.44.0-rc.0 (#15280)

Co-authored-by: andreancardona <andreancardona@users.noreply.github.com>

harukikuri committed:
fix(react): column span prop with percentage unit (#15223) (#15248)

* fix(react): column span prop with percentage unit (#15223)

* fix(react): column span prop with percentage unit (#15223)

* fix(react): column span prop with percentage unit (#15223)

guidari committed:
Added new drag an drop example using vite (#15235)

* feat: added new drag an drop using vite

* fix: removed yarn cache

* fix: revert yarn lock

* fix: removed image from vite

* chore: deleted old drag and drop example

renovate[bot] committed:
chore(deps): update actions/github-script action to v7 (#15229)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

joyful-these committed:
allow setting tabIndex of MenuButton (#15121)

* allow setting tabIndex of MenuButton

* update props validation

* chore: update snapshot

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

danoro96 committed:
refactor: copied propTypes comments

tw15egan committed:
feat(Tile): add `slug` prop to `SelectableTile` and `ExpandableTile` (#15222)

* docs(Tile): scaffold slug story, add demo styles

* feat(Tile): add slug to SelectableTile, ExpandableTile

* test(snapshot): update snapshots

* feat(Tile): add slug prop to Tile

* test(snapshot): update snapshots

* feat(Tile): add new tokens, add hover, selected styles

* test(snapshot): update snapshots

* feat(Tile): add new prop for rounded corners

* style(Tile): fix border-radius on pseudo elements

guidari committed:
fix: added indicators to fileuploader (#15274)

andreancardona committed:
feat: update react-router using vite (#15281)

* feat: update react-router with tabs

* feat: update react-router with tabs

* feat: cleaned up working example

* Update packages/react/examples/react-router-vite/README.md

Co-authored-by: Alison Joseph <alisonejoseph@Gmail.com>

* Update packages/react/examples/react-router-vite/README.md

Co-authored-by: Alison Joseph <alisonejoseph@Gmail.com>

---------

Co-authored-by: Alison Joseph <alisonejoseph@Gmail.com>

tw15egan committed:
feat(RadioButton): add `slug` prop (#15239)

* feat(RadioButton): add slug to RadioButtonGroup

* feat(RadioButton): add slug to RadioButton

* test(snapshot): update snapshots

* docs(Storybook): update tests stories for RadioButton

* feat(Checkbox): add slug styles to Checkbox

* test(snapshot): update snapshots

* refactor(Slug): remove whitespace

* style(RadioButton): use larger slug size when inline

alisonjoseph committed:
docs(examples): migrate custom data table state manager to vite (#15275)

* docs(examples): migrate custom data table state manager to vite

* chore: fix readme

* removed warning by changing the size name

---------

Co-authored-by: Guilherme Datilio Ribeiro <guilhermedatilio@gmail.com>

guidari committed:
fix: fixed typeError when sorting arrays (#15288)

lux01 committed:
refactor(Content): convert Content.js to TypeScript (#15133)

* refactor(Content): convert Content.js to TypeScript

* Update packages/react/src/components/UIShell/Content.ts

* Update packages/react/src/components/UIShell/Content.ts

* chore: yarn format

---------

Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>

cesardlinx committed:
fix: remove flicker in AccordionItem (#9818) (#14776)

* fix: remove flicker in AccordionItem (#9818)

Removes the flickering effect when using AccordionItem as a controlled
component

* fix: rework accordion component animation (#9818)

* refactor: use ref hook for accordion content

* test: update accordion component snapshots

* docs: fix extra missing curly brace in .all-contributorsrc

* docs: remove accordion controlled test story

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>

carbon-automation[bot] committed:
chore(release): v11.44.0 (#15292)

Co-authored-by: andreancardona <andreancardona@users.noreply.github.com>

guidari commented on packages/react/src/components/UIShell/HeaderGlobalAction.tsx:
```diff
@@ -0,0 +1,136 @@
+  /**
+   * Provide a custom icon for this global action
+   */
+  children?: ReactNode;
```
This one should be required
```suggestion
  children: ReactNode;
```

guidari reviewed (approved):
LGTM! 🚀 @danoro96 
Just one minor change

dependabot[bot] committed:
chore(deps): bump @adobe/css-tools from 4.2.0 to 4.3.2 (#15295)

Bumps [@adobe/css-tools](https://github.com/adobe/css-tools) from 4.2.0 to 4.3.2.
- [Changelog](https://github.com/adobe/css-tools/blob/main/History.md)
- [Commits](https://github.com/adobe/css-tools/commits)

---
updated-dependencies:
- dependency-name: "@adobe/css-tools"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alissa9090 committed:
refactor(Layer, useLayer): add TypeScript types (#15250)

* refactor(Layer, useLayer): add TypeScript types

* refactor: remove LayerLevel type for better readability

* refactor(Layer): add TypeScript types

* refactor(Layer): use BaseComponent

* refactor(Layer): add default value to as

tw15egan committed:
style(Fluid): fix issue with slug in fluid components, add fluid listbox (#15237)

* style(Fluid): fix issue with slug in fluid components, add fluid listbox

* test(e2e): fix a11y violations

danoro96 committed:
Make children required for HeaderGlobalAction.tsx

Co-authored-by: Guilherme Datilio Ribeiro <guilhermedatilio@gmail.com>

tay1orjones committed:
docs(storybook): remove outdated/deprecated addons (#15293)

* docs(storybook): remove outdated/deprecated addons

* chore(storybook): remove unused storybook-readme

conradennis committed:
feat(icons): icons metadata september thru november  (#15231)

* feat(icons): icons metadata september thru november

Added friendly names and aliases to icons.yml file

Removed foundation-model.svg from the repo, icons.yml and categories.yml files as requested by Will Burroughs

* test(snapshot): update snapshots

* chore: update snapshots

* chore: update snapshots

---------

Co-authored-by: TJ Egan <tw15egan@gmail.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency node to 20.10 (#15308)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

tw15egan committed:
feat(DataTable): add `Slug` to `DataTable` variants  (#15276)

* feat(DataTable): add TableSlugRow component

* style(DataTable): add more styles to DataTable with slug

* chore(DataTable): update stories, tweak styles

* style(DataTable): fix expandable non slug row hover styles

* style(DataTable): remove flicker from expandable hover

* style(DataTable): fix hover issues

* feat(DataTable): add slug to table header

* style(DataTable): adjust column styles

* test(snapshot): update snapshots

* feat(DataTable): add column header gradient

* chore(Storybook): move data table slug stories to experimental section

* style(DataTable): adjust slug in sort variant

guidari committed:
fix: fixed TypeError of undefined (#15313)

tw15egan committed:
style(Notification): fix issue with alignment when inline (#15314)

m4olivei committed:
feat(slider): add tooltips for single and two handle sliders (#15129)

* fix(slider): refine what is focusable to be more broadly inclusive

* feat(slider): add tooltips to slider

* fix(slider): adjust for nice default tooltips on single handle

* fix(slider): correct single handle RTL bug

* fix(slider): put back the focus handler for the lower thumb

* fix(slider): rtl positioning bug

* chore: udpate tests

* test(slider): adjust unit tests for slider w/ tooltips

* fix(slider): remove space when text input is hidden

* fix(slider): restore blue focus filled track

* feat(slider): adjustments to tooltip positioning

* feat(slider): use a ThumbWrapper component to conditionaly do tooltips

* feat(slider): add stories for hidden inputs

* fix(tooltip): hold the tooltip open if the user is mid-drag interaction

* refactor(slider): move slider handle componets into own file

* fix(slider): make skeletons good again

* fix(slider): manage focus when activating using the track

* chore(slider): remove redundant preventDefault

* fix(slider): more robust activeHandle detection

* test(slider): fix slider tests

* fix(slider): fix positioning of thumbs for rtl

* fix(slider): correct tooltips rtl

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

matejoslav committed:
feat(react): enable textarea to limit text input to set amount of words (#12906)

* feat(react): enable textarea to limit text input to set amount of words

Co-authored-by: SamChinellato <49278203+SamChinellato@users.noreply.github.com>
Co-authored-by: stevenpatrick009 <106097350+stevenpatrick009@users.noreply.github.com>

* test(textArea): fix tests and component

Fix tests affected by the merge to properly use the selected testing library.
Change TextArea component to remove duplicated use of the useRef hook.

* docs(developer handbook): change max characters for line in commit body

* fix(docs): regenerated contributors to resolve merge conflict

* revert sync file changes

* fix(TextArea): fixed eslint complaints

* refactor(TextArea): changes to counterMode and storybook

changed wording for counterMode to be character or word
removed storybook entry

* refactor(TextArea): removed unused test

* docs(README): added contributors

* revert(package.json & developer-handbook.md): reverted 90 limit to 80

* chore(format): run yarn format

* fix(README): fixed the contributors formatting

* refactor(TextArea): function to get initial text count

* refactor(useAnnouncer): made entityName have default value

* refactor(TextArea): use function to calculate text count in useEffect

* fix(TextArea): ignore max count limit when counter is disabled

* fix: fixed contributors list

* chore(TextArea): trim words on paste and add missing tests

* chore(TextArea): remove console log

* chore: Update packages/react/src/components/TextArea/__tests__/TextArea-test.js

Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>

* chore: remove redundant event definition in test

* fix(TextArea): fix count not updating when switching mode

* fix(TextArea): fix onPaste function

* fix(TextArea): fix lint errors and tests logic

* fix(contributors): revert list to what is on main

* fix: add contributors

---------

Co-authored-by: SamChinellato <49278203+SamChinellato@users.noreply.github.com>
Co-authored-by: stevenpatrick009 <106097350+stevenpatrick009@users.noreply.github.com>
Co-authored-by: TJ Egan <tw15egan@gmail.com>
Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>

alisonjoseph committed:
fix(menu): update styles (#15236)

* fix(menu): update spacing and add title to menuitem

* fix(menu): add max-height and scroll to menu

* fix: set hover color menu to primary

* chore: remove max-height value and revert storybook changes

* chore: storybook example

alisonjoseph committed:
fix(tabs): guard against current tab being undefined (#15317)

* fix(tabs): guard against current tab being undefined

* chore: delete testing story

danoro96 committed:
refactor: Add header panel types (#15096)

* refactor: add header panel types

* refactor: Fix README.md

* refactor: remove extra list start

* made props optional

* fix(headerpanel): update typings, add comments

---------

Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>
Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>

renovate[bot] committed:
chore(deps): update dependency lerna to v8 (#15309)

* chore(deps): update dependency lerna to v8

* chore: yarn dedupe

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>

tw15egan committed:
docs(Slug): move slug examples under experimental section (#15294)

* docs(Slug): move slug examples under experimental section

* refactor(Slug): move all slug examples under slug

* test(e2e): update form test to match new location

* refactor(Slug): add controls to slug example components

* docs(Slug): add slug to controls table

* style(TextArea): fix label width issue

alissa9090 committed:
refactor(InlineLoading): add TypeScript types (#15247)

* refactor(InlineLoading): add TypeScript types

* refactor(InlineLoading): add types and fix usages

* fix(inlineloading): name prop spread ...rest

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>

carbon-automation[bot] committed:
chore(release): v11.45.0-rc.1 (#15325)

Co-authored-by: alisonjoseph <alisonjoseph@users.noreply.github.com>

danoro96 committed:
refactor: add HeaderMenu.js back in

danoro96 committed:
refactor: re-remove HeaderMenu

harukikuri committed:
fix(react): popover caret is not hidden when caret prop is false (#15232)

* fix(react): popover caret is not hidden when caret prop is false

* fix(react): popover caret is not hidden when caret prop is false

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

dabrad26 committed:
fix(#15332) use toArray or checks for mapping children (#15333)

* fix(#15332) use toArray for mapping children

* fix: Content switcher

* fix: progress indicator

* fix: typescript rules for ReactElement usage

* fix: format diff test

xalc committed:
fix(bug): #15291 wrong selectedItems count (#15307)

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

tw15egan committed:
feat(Tag): add slug prop to Tag (#15318)

* feat(Tag): add slug prop to Tag

* style(Tag): change tag size, add margin

* refactor(Slug): guard against non-slug elements in slug slot

carbon-automation[bot] committed:
chore(release): v11.45.0 (#15343)

Co-authored-by: guidari <guidari@users.noreply.github.com>

tw15egan committed:
fix(Slug): ensure all necessary styles are imported (#15345)

Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump vite in /packages/react/examples/react-router-vite (#15331)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.0.3 to 5.0.5.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.0.5/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump vite (#15329)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.0.2 to 5.0.5.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.0.5/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tw15egan committed:
fix(Accordion): correctly set height when open prop is set (#15344)

janhassel committed:
fix(column): remove percentage character only if present (#15312)

Co-authored-by: Guilherme Datilio Ribeiro <guilhermedatilio@gmail.com>

kSee04 committed:
feat: paginationNav support high page counts (#15219)

* feat: paginationNav support high page counts

This commit updates PaginationNav to support high page counts.
I add a prop to PaginationNav which short circuits logic causing present issue

* Update README.md

* fix: code review

* docs(paginationnav): remove test story

---------

Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>
Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Taylor Jones <taylor.jones826@gmail.com>

alisonjoseph committed:
chore: update release docs and deploy-packages workflow (#15371)

* chore: update release docs and deploy-packages workflow

* docs(release): correct dist-tag syntax

---------

Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>

tw15egan committed:
feat(Modal): add slug to Modal, ComposedModal (#15350)

* feat(Modal): add slug to Modal, ComposedModal

* test(snapshot): update snapshots

* docs(Slug): add slug to controls table

* fix(Modal): remove shadows, use callout gradient

* test(snapshot): update snapshots

* fix(Modal): adjust background, fix conditional class

tay1orjones commented on packages/react/src/components/UIShell/HeaderMenu.tsx:
```diff
@@ -201,25 +264,23 @@ class HeaderMenu extends React.Component {
+      if (React.isValidElement(child)) {
+        child.props.isActive || child.props.isCurrentPage;
+      }
+    });
```
This is a functional change, right? This no longer recursively looks through descendant children. I think we'll want to retain that.

alisonjoseph committed:
feat(icons): rename infinity to infinity-symbol (#15384)

dependabot[bot] committed:
chore(deps): bump actions/configure-pages from 3 to 4 (#15321)

Bumps [actions/configure-pages](https://github.com/actions/configure-pages) from 3 to 4.
- [Release notes](https://github.com/actions/configure-pages/releases)
- [Commits](https://github.com/actions/configure-pages/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/configure-pages
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dudley-ibm committed:
feat(Pictograms): new productive master 231205 (#15328)

francinelucca committed:
Chore: Onboard @carbon/react to IBM Telemetry 🚀 (#15390)

* chore: add telemetry.yml config file

* chore: remove HTML attr from telemetry.yml, add ref to React attrs

* chore: onboard @carbon/react to IBM telemetry

* chore: add telemetry.yml to files

* fix: telemetry.yml path correction

* fix: sort telemetry.yml keys and wrap around single quotes

* fix: point schema version to v1

carbon-automation[bot] committed:
chore(release): v11.46.0-rc.0 (#15394)

* chore(release): v11.46.0-rc.0

* chore: update packagejson merge fix

* chore: yarn lock merge fix

---------

Co-authored-by: alisonjoseph <alisonjoseph@users.noreply.github.com>
Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>

dependabot[bot] committed:
chore(deps): bump actions/stale from 8 to 9 (#15353)

Bumps [actions/stale](https://github.com/actions/stale) from 8 to 9.
- [Release notes](https://github.com/actions/stale/releases)
- [Changelog](https://github.com/actions/stale/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/stale/compare/v8...v9)

---
updated-dependencies:
- dependency-name: actions/stale
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tay1orjones committed:
chore: remove node-gyp (#15369)

carbon-automation[bot] committed:
chore(release): v11.46.0 (#15400)

Co-authored-by: alisonjoseph <alisonjoseph@users.noreply.github.com>

guidari committed:
[Bug] - Sidenav menu collapse (#15349)

* fix: fixed menu open state

* test: adding avt test

* test: added test to open menu

* fix: removed comment

---------

Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>

dependabot[bot] committed:
chore(deps): bump github/codeql-action from 2 to 3 (#15393)

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2 to 3.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v2...v3)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tay1orjones committed:
fix(progressindicator): correct padding on optional labels in firefox (#15385)

Tresau committed:
feat: add TypeScript types for ErrorBoundary and ErrorBoundaryContext (#15348)

Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>

janhassel committed:
fix(treeview): prevent node selection on toggle (#15354)

tay1orjones committed:
docs(storybook): use storybook-addon-accessibility-checker (#14926)

* docs(storybook): use storybook-addon-accessibility-checker

* chore: yarn dedupe

* chore: update storybook-addon-accessibility-checker

* chore: yarn dedupe

* chore(yarn): run dedupe

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: TJ Egan <tw15egan@gmail.com>

guidari committed:
Added collapse announce in Expandable Search (#15355)

* fix: added collapse announce

* test: updated snapshots

* fix: fixed tests

* test: update snapshots and test

* test: fixed tests

* test: fixed test

---------

Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>
Co-authored-by: TJ Egan <tw15egan@gmail.com>

dependabot[bot] committed:
chore(deps): bump actions/setup-node from 4.0.0 to 4.0.1 (#15413)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 4.0.0 to 4.0.1.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/8f152de45cc393bb48ce5d89d36b731f54556e65...b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump vite (#15330)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.0.2 to 5.0.5.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.0.5/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

wkeese committed:
fix: declarations of ContentSwitcher, Slider, SkeletonIcon (#15382)

Addresses errors like:

TS2305: Module '"@carbon/react"' has no exported member 'ContentSwitcher'.
12 import { ContentSwitcher } from "@carbon/react";

This contains two types of fixes:

Whenever the underlying component is written in Typescript, the index file needs
to be named index.ts. If it's named index.js, the build won't even generate an
index.d.ts file. (The index file can also be named index.tsx, but that doesn't
make sense unless it contains React markup.)

I updated createClassWrapper() so that Typescript knows the type of the returned
functional component:

export function createClassWrapper<Props>(
  Component: ComponentClass<Props>
): FunctionComponent<Props> {
  ...
}

Note that the generated index.d.ts are a bit weird, for example

/// <reference types="react" />
declare const Slider:
   import("react").FunctionComponent<import("./Slider").SliderProps>;

If I add casting to the index.ts files themselves, ex:

const Slider = createClassWrapper(SliderComponent) as FC<SliderProps>;

That generates more canonical index.d.ts files:

import { FC } from "react";
import { SliderProps } from './Slider';
declare const Slider: FC<SliderProps>;

(So I can do that if you like.)

Of course, the createClassWrapper() calls should be temporary, as all remaining
class components should be converted to functional components.

Refs #13550, #12557, #13574.

dependabot[bot] committed:
chore(deps): bump actions/deploy-pages from 2 to 3 (#15322)

Bumps [actions/deploy-pages](https://github.com/actions/deploy-pages) from 2 to 3.
- [Release notes](https://github.com/actions/deploy-pages/releases)
- [Commits](https://github.com/actions/deploy-pages/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/deploy-pages
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: TJ Egan <tw15egan@gmail.com>

guidari committed:
`OverflowMenu` fix when oppening two menus (#15375)

* fix: fixed overflow menu with an id

* fix: fixed overflow

* fix: added first-child selector

* fix: removed duplicated story

---------

Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>
Co-authored-by: TJ Egan <tw15egan@gmail.com>

francinelucca committed:
fix: install @ibm/telemetry-js as a direct dependency (#15425)

tw15egan committed:
fix(Accordion): resolve overflow issue (#15423)

* style(Accordion): fix issue with content overflowing when collapsed

* style(Accordion): ensure tooltips can overflow, fix padding

dependabot[bot] committed:
chore(deps): bump actions/deploy-pages from 3 to 4 (#15434)

Bumps [actions/deploy-pages](https://github.com/actions/deploy-pages) from 3 to 4.
- [Release notes](https://github.com/actions/deploy-pages/releases)
- [Commits](https://github.com/actions/deploy-pages/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/deploy-pages
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mattborghi committed:
Add Typescript types to Stack component (#15404)

* refactor: changing file extensions to .tsx for Stack component

* refactor(typescript): add types to Stack component

mattborghi committed:
refactor(typescript): convert ModalWrapper to Typescript (#15359)

Co-authored-by: Guilherme Datilio Ribeiro <guilhermedatilio@gmail.com>

dev-ghostly committed:
fix: modified _number-input.scss:91 (#15421)

* fix: modified _number-input.scss:91

* fix: put back to 128px

* fix: adjust padding on invalid state

* fix: put back 128px

tw15egan committed:
fix(Textarea): ensure onKeyDown is called (#15419)

* fix(TextArea): ensure onKeyDown is called

* test(TextArea): add onKeyDown test

renovate[bot] committed:
fix(deps): update dependency got to v14 (#15411)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mattborghi committed:
Add Typescript types to PrimaryButton and SecondaryButton (#15405)

* refactor: change file extensions for SecondaryButton

* refactor: add types for SecondaryButton

* refactor: add types for PrimaryButton

alisonjoseph committed:
chore: remove carbon-components channel from release docs (#15401)

danoro96 committed:
refactor: undid previous changes and added element check

danoro96 committed:
Revert "refactor: undid previous changes and added element check"

This reverts commit d3be6d97caf3478461126973a3a35c33888c8b44.
